### ISSUE TITLE
Sendable Chooser returns default option despite correct option selected

### ISSUE THREAD

**Issue by VincentZhao12** [wpilibsuite/shuffleboard#775]
*Note this is all from memory as our team hasn't been back in the lab since this issue occurred*

We set up our SendableChooser<Command> as WPILib does in their [CommandBased Choosing an Autonomous Program](https://docs.wpilib.org/en/stable/docs/software/dashboards/smartdashboard/choosing-an-autonomous-program-from-smartdashboard.html) tutorial. Throughout our competition, we've been able to properly select autonomous routines this way, except for two matches. During those matches, the robot ran the default option of the SendableChooser rather than the selected path. When tested in pits, the SendableChooser appeared with a checkmark next to it however it still ran the default path rather than the selected one. No code was changed between matches before issue occurred. Returning the command we wanted directly from where we said autonSelector.getSelected() solved the issue, however we don't want to have to resort to using that and switching out between every match. 

WPILib Java, Version 2023.4.2
Roborio Version 2023_v3.2

Shuffleboard Log files from around the time that the issue occurred. 

[shuffleboard.2023-03-19-15.39.47.log](https://github.com/wpilibsuite/shuffleboard/files/11027428/shuffleboard.2023-03-19-15.39.47.log)
[shuffleboard.2023-03-19-15.39.18.log](https://github.com/wpilibsuite/shuffleboard/files/11027429/shuffleboard.2023-03-19-15.39.18.log)

Match Videos from about when the issue occurred: 
[First Occurrence](https://www.youtube.com/watch?v=Fug9MSlhTUg)
[Second Occurrence](https://www.youtube.com/watch?v=UJ4K7MNEe7U)

Match Video of expected behavior, one match before the issue occurred: 
[Video](https://www.youtube.com/watch?v=0KHZs37Xf04)

Match Video of just returning the command rather than the result of the SendableChooser:
[Video](https://www.youtube.com/watch?v=YjLkIRv1k9g)

**Comment by PeterJohnson** [comment]
I need to double-check to see if we aren’t already doing this, but Shuffleboard should set the “retained” property on the Sendable chooser’s selected entry so it is not unpublished if the dashboard disconnects, which is what sounds like may be happening.  We could also consider doing this on the robot code side.

**Comment by VincentZhao12** [comment]
Yes, once we get back in the lab we will try to recreate the issue just to make sure it wasn't somehow a robot code thing, but I'm very confident we didn't change anything code wise between the match where it worked vs the match where it didn't.

**Comment by jasondaming** [comment]
Another note here is that we also use FRC-web-components and the sendable chooser there has the same issue.

**Comment by Starlight220** [comment]
Does this reproduce with the newly-releases 2023.4.3?
